### PR TITLE
test: Port read_group query_tests to end-to-end tests

### PR DESCRIPTION
**Pull request by carols10cents**
Connects to #6617. Whew.

### CODE REVIEW

File: influxdb_iox/tests/end_to_end_cases/querier/influxrpc/read_group.rs [hunk 1]
```diff
@@ -369,3 +372,799 @@ async fn do_read_group_request(
 
     dump_data_frames(&frames)
 }
+
+#[tokio::test]
```
Comment (carols10cents) [inline influxdb_iox/tests/end_to_end_cases/querier/influxrpc/read_group.rs:376]: This test and the next test were`test_read_group_data_no_tag_columns` https://github.com/influxdata/influxdb_iox/blob/55257b46c942df9daea41bfaf16830c3fa218158/query_tests/src/influxrpc/read_group.rs#L60-L93

I decided to split them into two tests

File: influxdb_iox/tests/end_to_end_cases/querier/influxrpc/read_group.rs [hunk 2]
```diff
@@ -369,3 +372,799 @@ async fn do_read_group_request(
...
+            "GroupFrame, tag_keys: _field,_measurement, partition_key_vals: ",
+            "SeriesFrame, tags: _field=foo,_measurement=m0, type: 0",
+            "FloatPointsFrame, timestamps: [1], values: \"1\"",
+        ],
+    })
+    .run()
+    .await;
+}
+
+#[tokio::test]
```
Comment (carols10cents) [inline influxdb_iox/tests/end_to_end_cases/querier/influxrpc/read_group.rs:410]: Was `test_read_group_data_pred` https://github.com/influxdata/influxdb_iox/blob/55257b46c942df9daea41bfaf16830c3fa218158/query_tests/src/influxrpc/read_group.rs#L96-L116

File: influxdb_iox/tests/end_to_end_cases/querier/influxrpc/read_group.rs [hunk 3]
```diff
@@ -369,3 +372,799 @@ async fn do_read_group_request(
...
+            "GroupFrame, tag_keys: _field,_measurement,city,state, partition_key_vals: CA",
+            "SeriesFrame, tags: _field=temp,_measurement=h2o,city=LA,state=CA, type: 0",
+            "FloatPointsFrame, timestamps: [200], values: \"90\"",
+        ],
+    })
+    .run()
+    .await;
+}
+
+#[tokio::test]
```
Comment (carols10cents) [inline influxdb_iox/tests/end_to_end_cases/querier/influxrpc/read_group.rs:429]: was  `test_read_group_data_field_restriction` https://github.com/influxdata/influxdb_iox/blob/55257b46c942df9daea41bfaf16830c3fa218158/query_tests/src/influxrpc/read_group.rs#L119-L140

File: influxdb_iox/tests/end_to_end_cases/querier/influxrpc/read_group.rs [hunk 4]
```diff
@@ -369,3 +372,799 @@ async fn do_read_group_request(
...
+            "GroupFrame, tag_keys: _field,_measurement,city,state, partition_key_vals: MA",
+            "SeriesFrame, tags: _field=temp,_measurement=h2o,city=Boston,state=MA, type: 0",
+            "FloatPointsFrame, timestamps: [250], values: \"142.8\"",
+        ],
+    })
+    .run()
+    .await;
+}
+
+#[tokio::test]
```
Comment (carols10cents) [inline influxdb_iox/tests/end_to_end_cases/querier/influxrpc/read_group.rs:449]: Was `test_grouped_series_set_plan_sum` https://github.com/influxdata/influxdb_iox/blob/55257b46c942df9daea41bfaf16830c3fa218158/query_tests/src/influxrpc/read_group.rs#L143-L176

File: influxdb_iox/tests/end_to_end_cases/querier/influxrpc/read_group.rs [hunk 5]
```diff
@@ -369,3 +372,799 @@ async fn do_read_group_request(
...
+            "FloatPointsFrame, timestamps: [400], values: \"141\"",
+            "SeriesFrame, tags: _field=temp,_measurement=h2o,city=Cambridge,state=MA, type: 0",
+            "FloatPointsFrame, timestamps: [200], values: \"163\"",
+        ],
+    })
+    .run()
+    .await;
+}
+
+#[tokio::test]
```
Comment (carols10cents) [inline influxdb_iox/tests/end_to_end_cases/querier/influxrpc/read_group.rs:476]: Was `test_grouped_series_set_plan_count` https://github.com/influxdata/influxdb_iox/blob/55257b46c942df9daea41bfaf16830c3fa218158/query_tests/src/influxrpc/read_group.rs#L179-L212

File: influxdb_iox/tests/end_to_end_cases/querier/influxrpc/read_group.rs [hunk 6]
```diff
@@ -369,3 +372,799 @@ async fn do_read_group_request(
...
+            "IntegerPointsFrame, timestamps: [400], values: \"2\"",
+            "SeriesFrame, tags: _field=temp,_measurement=h2o,city=Cambridge,state=MA, type: 1",
+            "IntegerPointsFrame, timestamps: [200], values: \"2\"",
+        ],
+    })
+    .run()
+    .await;
+}
+
+#[tokio::test]
```
Comment (carols10cents) [inline influxdb_iox/tests/end_to_end_cases/querier/influxrpc/read_group.rs:505]: Was `test_grouped_series_set_plan_mean` https://github.com/influxdata/influxdb_iox/blob/55257b46c942df9daea41bfaf16830c3fa218158/query_tests/src/influxrpc/read_group.rs#L215-L246

File: influxdb_iox/tests/end_to_end_cases/querier/influxrpc/read_group.rs [hunk 7]
```diff
@@ -369,3 +372,799 @@ async fn do_read_group_request(
...
+            "FloatPointsFrame, timestamps: [400], values: \"70.5\"",
+            "SeriesFrame, tags: _field=temp,_measurement=h2o,city=Cambridge,state=MA, type: 0",
+            "FloatPointsFrame, timestamps: [200], values: \"81.5\"",
+        ],
+    })
+    .run()
+    .await;
+}
+
+#[tokio::test]
```
Comment (carols10cents) [inline influxdb_iox/tests/end_to_end_cases/querier/influxrpc/read_group.rs:530]: Was `test_grouped_series_set_plan_count_measurement_pred` https://github.com/influxdata/influxdb_iox/blob/55257b46c942df9daea41bfaf16830c3fa218158/query_tests/src/influxrpc/read_group.rs#L249-L277

File: influxdb_iox/tests/end_to_end_cases/querier/influxrpc/read_group.rs [hunk 8]
```diff
@@ -369,3 +372,799 @@ async fn do_read_group_request(
...
+            "GroupFrame, tag_keys: _field,_measurement,city,state, partition_key_vals: MA",
+            "SeriesFrame, tags: _field=temp,_measurement=h2o,city=Boston,state=MA, type: 1",
+            "IntegerPointsFrame, timestamps: [250], values: \"2\"",
+        ],
+    })
+    .run()
+    .await;
+}
+
+#[tokio::test]
```
Comment (carols10cents) [inline influxdb_iox/tests/end_to_end_cases/querier/influxrpc/read_group.rs:552]: Was `test_grouped_series_set_plan_first` https://github.com/influxdata/influxdb_iox/blob/55257b46c942df9daea41bfaf16830c3fa218158/query_tests/src/influxrpc/read_group.rs#L280-L307

File: influxdb_iox/tests/end_to_end_cases/querier/influxrpc/read_group.rs [hunk 9]
```diff
@@ -369,3 +372,799 @@ async fn do_read_group_request(
...
+            "IntegerPointsFrame, timestamps: [2000], values: \"7\"",
+            "SeriesFrame, tags: _field=s,_measurement=h2o,city=Cambridge,state=MA, type: 4",
+            "StringPointsFrame, timestamps: [2000], values: c",
+        ],
+    })
+    .run()
+    .await;
+}
+
+#[tokio::test]
```
Comment (carols10cents) [inline influxdb_iox/tests/end_to_end_cases/querier/influxrpc/read_group.rs:577]: Was `test_grouped_series_set_plan_first_with_nulls` https://github.com/influxdata/influxdb_iox/blob/55257b46c942df9daea41bfaf16830c3fa218158/query_tests/src/influxrpc/read_group.rs#L310-L341

File: influxdb_iox/tests/end_to_end_cases/querier/influxrpc/read_group.rs [hunk 10]
```diff
@@ -369,3 +372,799 @@ async fn do_read_group_request(
...
+            "FloatPointsFrame, timestamps: [250], values: \"70.4\"",
+            "SeriesFrame, tags: _field=temp,_measurement=h2o,city=Boston,state=MA, type: 0",
+            "FloatPointsFrame, timestamps: [50], values: \"70.4\"",
+        ],
+    })
+    .run()
+    .await;
+}
+
+#[tokio::test]
```
Comment (carols10cents) [inline influxdb_iox/tests/end_to_end_cases/querier/influxrpc/read_group.rs:606]: Was `test_grouped_series_set_plan_last` https://github.com/influxdata/influxdb_iox/blob/55257b46c942df9daea41bfaf16830c3fa218158/query_tests/src/influxrpc/read_group.rs#L344-L369

File: influxdb_iox/tests/end_to_end_cases/querier/influxrpc/read_group.rs [hunk 11]
```diff
@@ -369,3 +372,799 @@ async fn do_read_group_request(
...
+            "IntegerPointsFrame, timestamps: [3000], values: \"6\"",
+            "SeriesFrame, tags: _field=s,_measurement=h2o,city=Cambridge,state=MA, type: 4",
+            "StringPointsFrame, timestamps: [3000], values: b",
+        ],
+    })
+    .run()
+    .await;
+}
+
+#[tokio::test]
```
Comment (carols10cents) [inline influxdb_iox/tests/end_to_end_cases/querier/influxrpc/read_group.rs:631]: Was `test_grouped_series_set_plan_last_with_nulls` https://github.com/influxdata/influxdb_iox/blob/55257b46c942df9daea41bfaf16830c3fa218158/query_tests/src/influxrpc/read_group.rs#L372-L401

File: influxdb_iox/tests/end_to_end_cases/querier/influxrpc/read_group.rs [hunk 12]
```diff
@@ -369,3 +372,799 @@ async fn do_read_group_request(
...
+            "FloatPointsFrame, timestamps: [250], values: \"70.4\"",
+            "SeriesFrame, tags: _field=temp,_measurement=h2o,city=Boston,state=MA, type: 0",
+            "FloatPointsFrame, timestamps: [100000], values: \"70.4\"",
+        ],
+    })
+    .run()
+    .await;
+}
+
+#[tokio::test]
```
Comment (carols10cents) [inline influxdb_iox/tests/end_to_end_cases/querier/influxrpc/read_group.rs:657]: Was `test_grouped_series_set_plan_min` https://github.com/influxdata/influxdb_iox/blob/55257b46c942df9daea41bfaf16830c3fa218158/query_tests/src/influxrpc/read_group.rs#L404-L429

File: influxdb_iox/tests/end_to_end_cases/querier/influxrpc/read_group.rs [hunk 13]
```diff
@@ -369,3 +372,799 @@ async fn do_read_group_request(
...
+            "IntegerPointsFrame, timestamps: [3000], values: \"6\"",
+            "SeriesFrame, tags: _field=s,_measurement=h2o,city=Cambridge,state=MA, type: 4",
+            "StringPointsFrame, timestamps: [2000], values: a",
+        ],
+    })
+    .run()
+    .await;
+}
+
+#[tokio::test]
```
Comment (carols10cents) [inline influxdb_iox/tests/end_to_end_cases/querier/influxrpc/read_group.rs:682]: Was `test_grouped_series_set_plan_max` https://github.com/influxdata/influxdb_iox/blob/55257b46c942df9daea41bfaf16830c3fa218158/query_tests/src/influxrpc/read_group.rs#L432-L457

File: influxdb_iox/tests/end_to_end_cases/querier/influxrpc/read_group.rs [hunk 14]
```diff
@@ -369,3 +372,799 @@ async fn do_read_group_request(
...
+            "IntegerPointsFrame, timestamps: [2000], values: \"7\"",
+            "SeriesFrame, tags: _field=s,_measurement=h2o,city=Cambridge,state=MA, type: 4",
+            "StringPointsFrame, timestamps: [4000], values: z",
+        ],
+    })
+    .run()
+    .await;
+}
+
+#[tokio::test]
```
Comment (carols10cents) [inline influxdb_iox/tests/end_to_end_cases/querier/influxrpc/read_group.rs:707]: Was `test_grouped_series_set_plan_group_by_state_city` https://github.com/influxdata/influxdb_iox/blob/55257b46c942df9daea41bfaf16830c3fa218158/query_tests/src/influxrpc/read_group.rs#L460-L482

File: influxdb_iox/tests/end_to_end_cases/querier/influxrpc/read_group.rs [hunk 15]
```diff
@@ -369,3 +372,799 @@ async fn do_read_group_request(
...
+            MA,Cambridge",
+            "SeriesFrame, tags: _field=temp,_measurement=h2o,city=Cambridge,state=MA, type: 0",
+            "FloatPointsFrame, timestamps: [200], values: \"243\"",
+        ],
+    })
+    .run()
+    .await;
+}
+
+#[tokio::test]
```
Comment (carols10cents) [inline influxdb_iox/tests/end_to_end_cases/querier/influxrpc/read_group.rs:733]: Was `test_grouped_series_set_plan_group_by_city_state` https://github.com/influxdata/influxdb_iox/blob/55257b46c942df9daea41bfaf16830c3fa218158/query_tests/src/influxrpc/read_group.rs#L485-L508

File: influxdb_iox/tests/end_to_end_cases/querier/influxrpc/read_group.rs [hunk 16]
```diff
@@ -369,3 +372,799 @@ async fn do_read_group_request(
...
+            "FloatPointsFrame, timestamps: [600], values: \"21\"",
+            "SeriesFrame, tags: _field=temp,_measurement=h2o,city=LA,state=CA, type: 0",
+            "FloatPointsFrame, timestamps: [600], values: \"181\"",
+        ],
+    })
+    .run()
+    .await;
+}
+
+#[tokio::test]
```
Comment (carols10cents) [inline influxdb_iox/tests/end_to_end_cases/querier/influxrpc/read_group.rs:759]: Was `test_grouped_series_set_plan_group_aggregate_none` https://github.com/influxdata/influxdb_iox/blob/55257b46c942df9daea41bfaf16830c3fa218158/query_tests/src/influxrpc/read_group.rs#L511-L534

File: influxdb_iox/tests/end_to_end_cases/querier/influxrpc/read_group.rs [hunk 17]
```diff
@@ -369,3 +372,799 @@ async fn do_read_group_request(
...
+            "FloatPointsFrame, timestamps: [500, 600], values: \"10,11\"",
+            "SeriesFrame, tags: _field=temp,_measurement=h2o,city=LA,state=CA, type: 0",
+            "FloatPointsFrame, timestamps: [500, 600], values: \"90,91\"",
+        ],
+    })
+    .run()
+    .await;
+}
+
+#[tokio::test]
```
Comment (carols10cents) [inline influxdb_iox/tests/end_to_end_cases/querier/influxrpc/read_group.rs:786]: Was `test_grouped_series_set_plan_group_by_field_none` https://github.com/influxdata/influxdb_iox/blob/55257b46c942df9daea41bfaf16830c3fa218158/query_tests/src/influxrpc/read_group.rs#L537-L564

File: influxdb_iox/tests/end_to_end_cases/querier/influxrpc/read_group.rs [hunk 18]
```diff
@@ -369,3 +372,799 @@ async fn do_read_group_request(
...
+            "FloatPointsFrame, timestamps: [100], values: \"200.1\"",
+            "SeriesFrame, tags: _field=load2,_measurement=system,host=remote,region=B, type: 0",
+            "FloatPointsFrame, timestamps: [100, 200], values: \"2.1,20.2\"",
+        ],
+    })
+    .run()
+    .await;
+}
+
+#[tokio::test]
```
Comment (carols10cents) [inline influxdb_iox/tests/end_to_end_cases/querier/influxrpc/read_group.rs:820]: Was `test_grouped_series_set_plan_group_by_field_and_tag_none` https://github.com/influxdata/influxdb_iox/blob/55257b46c942df9daea41bfaf16830c3fa218158/query_tests/src/influxrpc/read_group.rs#L567-L598

File: influxdb_iox/tests/end_to_end_cases/querier/influxrpc/read_group.rs [hunk 19]
```diff
@@ -369,3 +372,799 @@ async fn do_read_group_request(
...
+            "FloatPointsFrame, timestamps: [100], values: \"200.1\"",
+            "SeriesFrame, tags: _field=load2,_measurement=system,host=local,region=C, type: 0",
+            "FloatPointsFrame, timestamps: [100], values: \"200.1\"",
+        ],
+    })
+    .run()
+    .await;
+}
+
+#[tokio::test]
```
Comment (carols10cents) [inline influxdb_iox/tests/end_to_end_cases/querier/influxrpc/read_group.rs:858]: Was `test_grouped_series_set_plan_group_by_tag_and_field_none` https://github.com/influxdata/influxdb_iox/blob/55257b46c942df9daea41bfaf16830c3fa218158/query_tests/src/influxrpc/read_group.rs#L601-L632

File: influxdb_iox/tests/end_to_end_cases/querier/influxrpc/read_group.rs [hunk 20]
```diff
@@ -369,3 +372,799 @@ async fn do_read_group_request(
...
```
Comment (carols10cents) [inline influxdb_iox/tests/end_to_end_cases/querier/influxrpc/read_group.rs:895]: Was `test_grouped_series_set_plan_group_measurement_tag_count` https://github.com/influxdata/influxdb_iox/blob/55257b46c942df9daea41bfaf16830c3fa218158/query_tests/src/influxrpc/read_group.rs#L635-L663

File: influxdb_iox/tests/end_to_end_cases/querier/influxrpc/read_group.rs [hunk 21]
```diff
@@ -369,3 +372,799 @@ async fn do_read_group_request(
...
+            "IntegerPointsFrame, timestamps: [100], values: \"1\"",
+            "SeriesFrame, tags: _field=load2,_measurement=system,host=local,region=C, type: 1",
+            "IntegerPointsFrame, timestamps: [100], values: \"1\"",
+        ],
+    })
+    .run()
+    .await;
+}
+
+#[tokio::test]
```
Comment (carols10cents) [inline influxdb_iox/tests/end_to_end_cases/querier/influxrpc/read_group.rs:931]: This test and the next one were `test_grouped_series_set_plan_group_field_start_stop` https://github.com/influxdata/influxdb_iox/blob/55257b46c942df9daea41bfaf16830c3fa218158/query_tests/src/influxrpc/read_group.rs#L666-L703

I decided to split them in two

File: influxdb_iox/tests/end_to_end_cases/querier/influxrpc/read_group.rs [hunk 22]
```diff
@@ -369,3 +372,799 @@ async fn do_read_group_request(
...
+            "IntegerPointsFrame, timestamps: [50], values: \"1\"",
+            "SeriesFrame, tags: _field=temp,_measurement=o2,city=Boston,state=MA, type: 1",
+            "IntegerPointsFrame, timestamps: [50], values: \"1\"",
+        ],
+    })
+    .run()
+    .await;
+}
+
+#[tokio::test]
```
Comment (carols10cents) [inline influxdb_iox/tests/end_to_end_cases/querier/influxrpc/read_group.rs:983]: Was `test_grouped_series_set_plan_group_field_pred_and_null_fields` https://github.com/influxdata/influxdb_iox/blob/55257b46c942df9daea41bfaf16830c3fa218158/query_tests/src/influxrpc/read_group.rs#L706-L732

File: influxdb_iox/tests/end_to_end_cases/querier/influxrpc/read_group.rs [hunk 23]
```diff
@@ -369,3 +372,799 @@ async fn do_read_group_request(
...
+            "IntegerPointsFrame, timestamps: [50], values: \"1\"",
+            "GroupFrame, tag_keys: _field,_measurement,city,state, partition_key_vals: MA,temp",
+            "SeriesFrame, tags: _field=temp,_measurement=o2,city=Boston,state=MA, type: 1",
+            "IntegerPointsFrame, timestamps: [50], values: \"1\"",
+        ],
+    })
+    .run()
+    .await;
+}
+
```
Comment (carols10cents) [inline influxdb_iox/tests/end_to_end_cases/querier/influxrpc/read_group.rs:1009]: Was `test_grouped_series_set_plan_group_field_pred_filter_on_field` https://github.com/influxdata/influxdb_iox/blob/55257b46c942df9daea41bfaf16830c3fa218158/query_tests/src/influxrpc/read_group.rs#L734-L763

File: influxdb_iox/tests/end_to_end_cases/querier/influxrpc/read_group.rs [hunk 24]
```diff
@@ -369,3 +372,799 @@ async fn do_read_group_request(
...
+            "IntegerPointsFrame, timestamps: [300], values: \"0\"",
+            "GroupFrame, tag_keys: _field,_measurement,city,state, partition_key_vals: MA,reading",
+            "SeriesFrame, tags: _field=reading,_measurement=o2,city=Boston,state=MA, type: 1",
+            "IntegerPointsFrame, timestamps: [50], values: \"1\"",
+        ],
+    })
+    .run()
+    .await;
+}
+
```
Comment (carols10cents) [inline influxdb_iox/tests/end_to_end_cases/querier/influxrpc/read_group.rs:1035]: Was `test_grouped_series_set_plan_group_field_pred_filter_on_value` https://github.com/influxdata/influxdb_iox/blob/55257b46c942df9daea41bfaf16830c3fa218158/query_tests/src/influxrpc/read_group.rs#L765-L796

File: influxdb_iox/tests/end_to_end_cases/querier/influxrpc/read_group.rs [hunk 25]
```diff
@@ -369,3 +372,799 @@ async fn do_read_group_request(
...
+            "GroupFrame, tag_keys: _field,_measurement,host, partition_key_vals: load4",
+            "SeriesFrame, tags: _field=load4,_measurement=system,host=host.local, type: 0",
+            "FloatPointsFrame, timestamps: [1527018806000000000], values: \"1.77\"",
+        ],
+    })
+    .run()
+    .await;
+}
+
+#[tokio::test]
```
Comment (carols10cents) [inline influxdb_iox/tests/end_to_end_cases/querier/influxrpc/read_group.rs:1060]: Was `test_grouped_series_set_plan_group_field_pred_filter_on_multiple_value` https://github.com/influxdata/influxdb_iox/blob/55257b46c942df9daea41bfaf16830c3fa218158/query_tests/src/influxrpc/read_group.rs#L799-L827

File: influxdb_iox/tests/end_to_end_cases/querier/influxrpc/read_group.rs [hunk 26]
```diff
@@ -369,3 +372,799 @@ async fn do_read_group_request(
...
```
Comment (carols10cents) [inline influxdb_iox/tests/end_to_end_cases/querier/influxrpc/read_group.rs:1084]: Was `test_grouped_series_set_plan_group_field_pred_filter_on_value_sum` https://github.com/influxdata/influxdb_iox/blob/55257b46c942df9daea41bfaf16830c3fa218158/query_tests/src/influxrpc/read_group.rs#L830-L856

File: influxdb_iox/tests/end_to_end_cases/querier/influxrpc/read_group.rs [hunk 27]
```diff
@@ -369,3 +372,799 @@ async fn do_read_group_request(
...
+            "GroupFrame, tag_keys: _field,_measurement,host, partition_key_vals: load4",
+            "SeriesFrame, tags: _field=load4,_measurement=system,host=host.local, type: 0",
+            "FloatPointsFrame, timestamps: [1527018826000000000], values: \"3.54\"",
+        ],
+    })
+    .run()
+    .await;
+}
+
+#[tokio::test]
```
Comment (carols10cents) [inline influxdb_iox/tests/end_to_end_cases/querier/influxrpc/read_group.rs:1105]: Was `test_read_group_with_periods` https://github.com/influxdata/influxdb_iox/blob/55257b46c942df9daea41bfaf16830c3fa218158/query_tests/src/influxrpc/read_group.rs#L859-L882

**Review by carols10cents — COMMENTED** [review]
Matching up tests

**Review by alamb — APPROVED** [review]
I went through these changes carefully and they look great -- the attention to detail is inspiring. Thank you so much @carols10cents .

**Comment by alamb** [comment]
I am taking some liberty and merging this PR to keep things moving along

**Pull request merged**